File: C version/array/removeDuplicates.c
#include <stdio.h>
int removeDuplicates(int* nums, int numsSize){
    int i = 0;
    int n = 0;
    for(;i<numsSize;i++){
        if(i==0 || nums[i]!=nums[i-1]){
            nums[n] = nums[i];
            n++;
        }
    }
    printf("%d %d",numsSize,n);
    return n;
}
void main(){
    int a[10]={0,0,1,1,1,2,2,3,3,4};
    removeDuplicates(a,sizeof(a)/4);
}

File: Rust-version/src/problem/remove_duplicates.rs
use super::Solution;

impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut cur_index = 0;
        for i in 1..nums.len() {
            if nums[i] != nums[cur_index] {
                cur_index += 1;
                nums[cur_index] = nums[i];
            }
        }
        cur_index as i32 + 1
    }
}